packages/frama-clang/frama-clang.0.0.15/opam

```diff
@@ -0,0 +1,46 @@
+opam-version: "2.0"
+synopsis: "Frama-C plug-in based on Clang for parsing C++ files"
+description:
+  "This Frama-C plug-in parse C++ files that may content ACSL++ annotations."
+maintainer: "Virgile.Prevosto@cea.fr"
+authors: "Virgile Prevosto"
+license: "LGPL-2.1-only"
+tags: ["formal specification" "C++" "plugins" "ACSL" "ACSL++"]
+homepage: "https://frama-c.com/fc-plugins/frama-clang.html"
+bug-reports: "https://git.frama-c.com/pub/frama-clang"
+depends: [
+  "dune" {>= "3.2"}
+  "frama-c" {>= "28.0~" & < "29.0~"}
+  "zarith" {>= "1.5"}
+  "camlp5"
+  "camlp-streams"
+  "conf-llvm" {>= "11.0.0"}
```

Comment:
Seems that the bound here is incorrect:

```
# Error: No usable version of LLVM <=12.0.x found.
```

See full log [here](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2df9d23bd233dc90a2117aba38ce6137c2374b28/variant/compilers,4.13,frama-clang.0.0.15,lower-bounds)

Reply:
Well, I'd argue that the bulk of the issue lies in `conf-libclang`, which apparently insist on installing v12 while the underlying system only offers llvm 13 or higher. we check frama-clang itself against llvm 11 to 17, but the problem is to convey this information through the `depext` layer. I don't know of a satisfactory solution, especially as `conf-llvm` and `conf-libclang` seem to differ as to which version they should advertise:

```
 - downgrade conf-libclang               15 to 12             [required by frama-clang]
 - downgrade conf-llvm                   16 to 13.0.0         [required by frama-clang]
```

A possibility would be not to rely on the version number of `conf-libclang` and `conf-llvm` and perform a check at configure stage.

NB: the [problem](https://github.com/ocaml/opam-repository/pull/24344#issuecomment-1706908423) was already present for frama-clang.0.0.14.